usermods/Internal_Temperature_v2/usermod_internal_temperature.h

```diff
@@ -41,6 +55,30 @@ class InternalTemperatureUsermod : public Usermod
     temperature = roundf(temperatureRead() * 10) / 10;
 #endif
 
+    // Check if temperature has gone above the threshold
+    if (temperature >= activationThreshold) {
+      // Update the state flag if not already set
+      if (!isAboveThreshold){
+        isAboveThreshold = true;
+        }
+      // Activate the 'over-threshold' preset if it's not already active
+      if (presetToActivate != -1 && currentPreset != presetToActivate) {
+        saveTemporaryPreset();  // Save the current preset to allow re-activation later
```

Comment:
This does not save *preset ID*. It only saves current state in temporary preset.